src/classes/storage.ts

```diff
@@ -84,4 +100,155 @@ export class Storage {
 
     this.groundPlaces[placeGpuid] = newGroundPlace;
   }
+
+  /**
+   * @description Allows you to filter the ground places according to criteria such as name, gpuid, etc.
+   * @param {AutoCompleteFilters[]|undefined} filters - The filters you want to be applied on the search.
+   * @returns {AutoComplete[]}
+   */
+  public filterGroundPlaces(filters?: AutoCompleteFilters[]): AutoComplete[] {
+    const groundPlacesAutocomplete: AutoComplete[] = this.parseGroundPlaces();
+
+    // Returns earlier the Ground places if there is no filters to use.
+    if (!filters || !filters.length) return groundPlacesAutocomplete;
+
+    const filteredByGroup: AutoComplete[] = this.filterGroundPlacesByGroup(
+      groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.STOP_GROUP),
+    );
+
+    const filteredByCluster: AutoComplete[] = this.filterGroundPlacesByCluster(
+      groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.STOP_CLUSTER),
+    );
+
+    const filteredByType: AutoComplete[] = [...filteredByGroup, ...filteredByCluster];
+
+    const filteredBySegmentProviderStop: AutoComplete[] = this.filterGroundPlacesBySegmentProvider(
+      // Filtering by segment provider must be made on the whole results previously filtered.
+      // We checks here if this filter must be applied on results already filtered or not.
+      filteredByType.length ? filteredByType : groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.SEGMENT_PROVIDER_STOP),
+    );
+
+    const filteredByService: AutoComplete[] = this.filterGroundPlacesByServiced(
+      filteredBySegmentProviderStop,
+      filters.includes(AutoCompleteFilters.SERVICED),
+    );
+
+    return filteredByService;
```

Comment:
This methode should be way much simpler

Reply:
I had done almost the same function on the front side when I was using the autocomplete. Is it really that hard to read?

Reply:
I guess there are other ways to implement filters that may overlap, but in vanilla JS applying filters to other filters can be implemented this way.

What's your opinion on this @rimiti @theo-le-donne